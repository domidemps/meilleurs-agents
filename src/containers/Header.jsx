/** @jsx jsx */
import {css, jsx} from '@emotion/core'
import AppBar from '@material-ui/core/AppBar'
import React from 'react'
import Toolbar from '@material-ui/core/Toolbar'

import AgencySelector from './AgencySelector'
import MeilleursAgentsLogo from '../assets/logo-meilleursagentspro-neg.svg'
import UnreadMessagesCounter from './UnreadMessagesCounter'
import media from '../styles/media'

const styles = css`
  display: flex;
  .toolBar {
    display: flex;
    justify-content: space-between;
  }
  .logo {
    vertical-align: middle;
  }
  ${media.phone`
    .logo {
      width: 150px;
    }
    .displayContents {
      display: contents;
    }
  `}
`

export default function Header() {
  return (
    <AppBar position="sticky" css={styles}>
      <Toolbar className="toolBar">
        <div className="displayContents">
          <img src={MeilleursAgentsLogo} alt="MeilleursAgents PRO" className="logo" />
          <UnreadMessagesCounter />
        </div>
        <AgencySelector />
      </Toolbar>
    </AppBar>
  )
}
