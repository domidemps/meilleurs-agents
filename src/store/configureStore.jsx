import {applyMiddleware, compose, createStore} from 'redux'
import {persistReducer, persistStore} from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import {routerMiddleware} from 'connected-react-router'
import promise from 'redux-promise'
import thunk from 'redux-thunk'

import reducers, {history} from '../reducers'

const middlewares = [thunk, promise, routerMiddleware(history)]

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['realtors', 'messages'],
}

const persistedRootReducer = persistReducer(persistConfig, reducers)
const createStoreWithEnhancers = composeEnhancers(applyMiddleware(...middlewares))(createStore)
export const store = createStoreWithEnhancers(persistedRootReducer)
export const persistor = persistStore(store)
