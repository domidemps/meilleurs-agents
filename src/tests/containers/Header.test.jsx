import React from 'react'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer'
import {createMemoryHistory} from 'history'
import {Router, Switch} from 'react-router-dom'

import Header from '../../containers/Header'

export const history = createMemoryHistory({initialEntries: ['/']})

describe('Header', () => {
  const store = configureStore([thunk])({
    realtors: {
      agencies: [
        {
          id: 101,
          logo: 'http://placehold.it/100x100?text=Agence+101',
          name: 'Agence #101',
          unread_messages: 73,
        },
        {
          id: 102,
          logo: 'http://placehold.it/100x100?text=Agence+102',
          name: 'Agence #102',
          unread_messages: 68,
        },
        {
          id: 103,
          logo: 'http://placehold.it/100x100?text=Agence+103',
          name: 'Agence #103',
          unread_messages: 76,
        },
      ],
      selectedAgency: '101',
      unreadMessages: 73,
    },
  })
  it('renders correctly for computers/tablets', () => {
    global.innerWidth = 600
    expect(
      renderer.create(
        <Provider store={store}>
          <Router history={history}>
            <Switch>
              <Header/>
            </Switch>
          </Router>
        </Provider>
      ).toJSON()
    ).toMatchSnapshot()
  })
  it('renders correctly for smartphone', () => {
    global.innerWidth = 350
    expect(
      renderer.create(
        <Provider store={store}>
          <Router history={history}>
            <Switch>
              <Header/>
            </Switch>
          </Router>
        </Provider>
      ).toJSON()
    ).toMatchSnapshot()
  })
})