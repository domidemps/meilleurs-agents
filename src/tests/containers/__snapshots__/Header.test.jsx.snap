// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly for computers/tablets 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary css-anjygb-Header MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular toolBar MuiToolbar-gutters"
  >
    <div>
      <img
        alt="MeilleursAgents PRO"
        className="logo"
        src={[Function]}
      />
      <div
        className="MuiChip-root css-1we3e77-UnreadMessagesCounter MuiChip-colorPrimary"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiChip-icon MuiChip-iconColorPrimary"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
        <span
          className="MuiChip-label"
        >
          <strong>
            73
          </strong>
        </span>
      </div>
    </div>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline css-kkkune-AgencySelector"
      onClick={[Function]}
    >
      <select
        className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
      >
        <option
          className="options"
          value={101}
        >
          Agence #101
        </option>
        <option
          className="options"
          value={102}
        >
          Agence #102
        </option>
        <option
          className="options"
          value={103}
        >
          Agence #103
        </option>
      </select>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiNativeSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</header>
`;

exports[`Header renders correctly for smartphone 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary css-anjygb-Header MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular toolBar MuiToolbar-gutters"
  >
    <div>
      <img
        alt="MeilleursAgents PRO"
        className="logo"
        src={[Function]}
      />
      <div
        className="MuiChip-root css-1we3e77-UnreadMessagesCounter MuiChip-colorPrimary"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiChip-icon MuiChip-iconColorPrimary"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
        <span
          className="MuiChip-label"
        >
          <strong>
            73
          </strong>
        </span>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiIconButton-root css-hna2n5-AgencySelector"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root css-15pd8ov-AgencySelector"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4v2h16V4zm1 10v-2l-1-5H4l-1 5v2h1v6h10v-6h4v6h2v-6h1zm-9 4H6v-4h6v4z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</header>
`;
